tools/release/update_setup_requirements.py

```diff
@@ -8,45 +8,87 @@
 # Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
 # an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
 # specific language governing permissions and limitations under the License.
+# --------------------------------------------------------------------------------------------------
+# Updates the setup.requirements.txt files for a given package.
+#
+# Invoked by workflows/build-and-release-single-package.yml and workflows/build-and-release.yml.
+# Working directory must be [root_dir].
+# --------------------------------------------------------------------------------------------------
 
 import os
+import re
 import sys
-from typing import Dict
+import typing as t
+
+from common import PACKAGES, retrieve_github_path
 
 BASE_PATH = "./tools/packages"
 
 
-def __build_taipy_package_line(line: str, version: str, publish_on_py_pi: bool) -> str:
-    _line = line.strip()
-    if publish_on_py_pi:
-        return f"{_line}=={version}\n"
-    tag = f"{version}-{_line.split('-')[1]}"
-    tar_name = f"{_line}-{version}"
-    return f"{_line} @ https://github.com/Avaiga/taipy/releases/download/{tag}/{tar_name}.tar.gz\n"
+def usage() -> None:
+    packages = "> <".join(f"{p}_ver" for p in PACKAGES)
+    print(  # noqa: T201
+        f"Usage: {sys.argv[0]} <package> <{packages}> <deps> [<gh_path>]"
+    )
+    packages = ", ".join(f"'{p}'" for p in PACKAGES[:-1])
+    packages = f"{packages}, or '{PACKAGES[-1]}'"
+    print(f"   <package> must be one of {packages}.")  # noqa: T201
+    for p in PACKAGES:
+        print(f"   <{p}_ver>: minimal version of the taipy-{p} dependency.")  # noqa: T201
+    print("   <deps> must be 'Pypi' or 'GitHub', indicating where to find Taipy package dependencies.")  # noqa: T201
+    print("   <gh_path>: The path of GitHub repository (owner/repo), used if <deps> is 'GitHub'.")  # noqa: T201
+
 
+def __build_taipy_package_line(line: str, version: str, use_pypi: bool, gh_path: t.Optional[str]) -> str:
+    line = line.strip()
+    if use_pypi:
+        # Target dependency version should the latest compatible with 'version'
+        major, minor = version.split(".")[:2]
+        return f"{line} >={major}.{minor},<{major}.{int(minor) + 1}\n"
+    tag = f"{version}-{line.split('-')[1]}"
+    tar_name = f"{line}-{version}"
+    return f"{line} @ https://github.com/{gh_path}/releases/download/{tag}/{tar_name}.tar.gz\n"
 
-def update_setup_requirements(package: str, versions: Dict, publish_on_py_pi: bool) -> None:
-    _path = os.path.join(BASE_PATH, package, "setup.requirements.txt")
+
+def update_setup_requirements(
+    package: str, versions: dict[str, str], publish_on_py_pi: bool, gh_path: t.Optional[str]
```

Comment:
Why not using the Version class?
```suggestion
    package: str, versions: dict[str, Version], publish_on_py_pi: bool, gh_path: t.Optional[str]
```